app/components/facilities-section.js

```diff
@@ -0,0 +1,158 @@
+import Ember from 'ember'; // eslint-disable-line
+import mapboxgl from 'mapbox-gl'; // eslint-disable-line
+
+import carto from '../utils/carto';
+
+const colorsArray = [
+  ['Education, Child Welfare, and Youth', '#f7ca00'],
+  ['Parks, Gardens, and Historical Sites', '#4CAF50'],
+  ['Libraries and Cultural Programs', '#73E5F4'],
+  ['Public Safety, Emergency Services, and Administration of Justice', '#2979FF'],
+  ['Health and Human Services', '#BA68C8'],
+  ['Core Infrastructure and Transportation', '#8D8EAA'],
+  ['Administration of Government', '#CBCBD6'],
+];
+
+export default Ember.Component.extend({
+  borocd: '',
+
+  init() {
+    this._super(...arguments);
+
+    const SQL = `
+      SELECT the_geom_webmercator, facname, facdomain, uid FROM facdb_facilities;
+    `;
+
+    carto.getTileTemplate(SQL)
+      .then((facilitiesTemplate) => {
+        this.set('facilitiesTemplate', facilitiesTemplate);
+      });
```

Comment:
Thinking about this a little more, I think this promise can just be moved to `vectorSource`. Since the resolution of the promise won't force a full redraw of the map component (`{{mapbox-gl}}`), we can use promise helpers to conditionally - based on when the promise resolves - yield the contextual components for source and layer:

```handlebars
{{#mapbox-gl}}
  {{#if (await vectorSource)}}
    {{mapbox-gl-source template=vectorSource}} // don't remember the API here, but you get the picture
  {{/if}}
{{/mapbox-gl}}
```